src/shared/bus-util.c

```diff
@@ -527,6 +527,7 @@ int bus_verify_polkit_async(
 
         r = sd_bus_call_async(call->bus, &q->slot, pk, async_polkit_callback, q, 0);
         if (r < 0) {
+                hashmap_remove(*registry, call);
                 async_polkit_query_free(q);
```

Comment:
not following… async_polkit_query_free() already includes the hashmap_remove() anyway, if both q->registry and q->request are initialized, which they are in this case. This means your addition is unnecessary, afaics?

What's the precise issue you are encountering, that you are trying to fix there?

Reply:
Eh, sorry. It was a PEBKAC.